catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/util/hardware/SensorTestArduinoServerConnection.java

```
@@ -0,0 +1,247 @@
+		if (expected == SET_LED_ON_VALUE) {
+			expectedChar = '1';
+			assertString = "Error: LED is turned off!";
+		} else {
+			expectedChar = '0';
+			assertString = "Error: LED is turned on!";
+		}
+		try {
+			connectToArduinoServer();
+			Thread.sleep(NETWORK_DELAY_MS);
+			Log.d(TAG, "requesting sensor value: ");
+			sendToServer.writeByte(Integer.toHexString(GET_LIGHT_VALUE_ID).charAt(0));
+			sendToServer.flush();
+			Thread.sleep(NETWORK_DELAY_MS);
+			response = receiveFromServer.readLine();
+			Log.d(TAG, "response received! " + response);
+			clientSocket.close();
+
+			assertFalse("Wrong Command!", response.contains("ERROR"));
+			assertTrue("Wrong data received!", response.contains("LIGHT_END"));
+			assertTrue(assertString, response.charAt(0) == expectedChar);
+			closeConnection();
+		} catch (IOException ioException) {
+			throw new AssertionFailedError("Data exchange failed! Check server connection!");
+		} catch (InterruptedException e) {
+			Log.w(TAG, "InterruptedException", e);
+		}
+	}
+
+	public static void checkAudioSensorValue(int expected) {
+		char expectedChar;
+		String assertString;
+		String response;
+		if (expected == SET_AUDIO_ON_VALUE) {
+			expectedChar = '1';
+			assertString = "Error: Audio is turned off!";
+		} else {
+			expectedChar = '0';
+			assertString = "Error: Audio is turned on!";
+		}
+		try {
+			connectToArduinoServer();
+			Thread.sleep(NETWORK_DELAY_MS);
+			Log.d(TAG, "requesting sensor value: ");
+			sendToServer.writeByte(Integer.toHexString(GET_AUDIO_VALUE_ID).charAt(0));
+			sendToServer.flush();
+			Thread.sleep(NETWORK_DELAY_MS);
+			response = receiveFromServer.readLine();
+			Log.d(TAG, "response received! " + response);
+			clientSocket.close();
+
+			assertFalse("Wrong Command!", response.contains("ERROR"));
+			assertTrue("Wrong data received!", response.contains("AUDIO_END"));
+			assertTrue(assertString, response.charAt(0) == expectedChar);
+			closeConnection();
+		} catch (IOException ioException) {
```

please remove try catch blocks and let the function throw exceptions...